core/block/object/objectcreator/relation_option.go

```diff
@@ -52,3 +55,40 @@ func (s *service) createRelationOption(ctx context.Context, space clientspace.Sp
 	setOriginalCreatedTimestamp(createState, details)
 	return s.CreateSmartBlockFromStateInSpace(ctx, space, []domain.TypeKey{bundle.TypeKeyRelationOption}, createState)
 }
+
+func (s *service) setOptionOrderId(details *domain.Details, spc clientspace.Space) {
+	records, err := s.objectStore.SpaceIndex(spc.Id()).Query(database.Query{
+		Filters: []database.FilterRequest{
+			{
+				RelationKey: bundle.RelationKeyResolvedLayout,
+				Condition:   model.BlockContentDataviewFilter_Equal,
+				Value:       domain.Int64(model.ObjectType_relationOption),
+			},
+			{
+				RelationKey: bundle.RelationKeyRelationKey,
+				Condition:   model.BlockContentDataviewFilter_Equal,
+				Value:       details.Get(bundle.RelationKeyRelationKey),
+			},
+			{
+				RelationKey: bundle.RelationKeyOrderId,
+				Condition:   model.BlockContentDataviewFilter_NotEmpty,
+			},
+		},
+		Sorts: []database.SortRequest{{
+			RelationKey: bundle.RelationKeyOrderId,
+			Type:        model.BlockContentDataviewSort_Asc,
+			NoCollate:   true,
+		}},
+		Limit: 1,
+	})
+
+	if err != nil {
+		log.With("spaceID", spc.Id()).Errorf("failed to query object types with orders to set orderId to new type: %v", err)
```

Comment:
As a rule of a thumb you should defer logging as long as you can, it keeps code purer (in a sense of side-effects)